eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -1072,11 +1088,60 @@ private ClientResponse makeRemoteCall(Action action) throws Throwable {
      * @throws Throwable
      *             on any error.
      */
-    private ClientResponse makeRemoteCall(Action action, int serviceUrlIndex)
-            throws Throwable {
+    private ClientResponse makeRemoteCall(Action action, int serviceUrlIndex) throws Throwable {
+        String serviceUrl;
+        try {
+            serviceUrl = eurekaServiceUrls.get().get(serviceUrlIndex);
+            return makeRemoteCallWithFollowRedirect(action, serviceUrl);
+        } catch (Throwable t) {
+            if (eurekaServiceUrls.get().size() > ++serviceUrlIndex) {
+                logger.warn("Trying backup: " + eurekaServiceUrls.get().get(serviceUrlIndex));
+                SERVER_RETRY_COUNTER.increment();
+                return makeRemoteCall(action, serviceUrlIndex);
+            } else {
+                ALL_SERVER_FAILURE_COUNT.increment();
+                logger.error("Can't contact any eureka nodes - possibly a security group issue?", t);
+                throw t;
+            }
+        }
+    }
+
+    private ClientResponse makeRemoteCallWithFollowRedirect(Action action, String serviceUrl) throws Throwable {
+        URI targetUrl = new URI(serviceUrl);
+        for(int followRedirectCount = 0; followRedirectCount < 10; followRedirectCount++) {
+            ClientResponse clientResponse = makeRemoteCall(action, targetUrl.toString());
+            if(clientResponse.getStatus() < 300) {
```

Comment:
Isn't the below code checking for redirects? Shouldn't this check be checking for code b/w 300-400 ?